site/content/extend/plugins/webapp/hello-world.md

```diff
@@ -1,66 +1,157 @@
+npm install --save-dev webpack webpack-cli babel-loader babel-core babel-preset-env babel-preset-react
+npm install --save react
+```
 
-To quickly set up a workspace, change into a directory where you want to store your plugins in development and run:
+Configure Babel by creating a `.babelrc` file:
 
+```json
+{
+  "presets": ["env", "react"]
+}
 ```
-mdk init plugin --name helloworld --components ProfilePopover --skip-prompts
+
+Configure Webpack by creating a `webpack.config.js` file:
+
+```js
+var path = require('path');
+var webpack = require('webpack');
+
+module.exports = {
+    entry: [
+        './src/index.jsx',
+    ],
+    resolve: {
+        extensions: ['*', '.js', '.jsx']
+    },
+    module: {
+        rules: [
+            {
+                test: /\.(js|jsx)$/,
+                exclude: /node_modules/,
+                use: 'babel-loader',
+            },
+        ],
+    },
+    externals: {
+        react: 'react',
+    },
+    output: {
+        path: path.join(__dirname, '/dist'),
+        publicPath: '/',
+        filename: 'main.js'
+    },
+};
 ```
 
-This will create your plugin workspace in a directory called `helloworld`.
+Observe that `react` is specified as an external library. This allows you to test your code locally (e.g. with [jest](https://jestjs.io/) and snapshots) but leverage the version of React shipped with Mattermost to avoid bloating your plugin.
 
-In the future, you can run `mdk init plugin` to have prompts walk you through the setup.
+Now create the entry point file and output directory:
+```bash
+mkdir src dist
+touch src/index.js
+```
 
-## Building the Plugin
+Then populate `src/index.jsx` with the following:
+```js
+import React from 'react'
+
+// Courtesy of https://feathericons.com/
+const ApertureIcon = () => (
+    <svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="-3 -3 30 30" fill="white" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-aperture">
+        <circle cx="12" cy="12" r="10" fill="none"/></circle>
+        <line x1="14.31" y1="8" x2="20.05" y2="17.94"></line>
+        <line x1="9.69" y1="8" x2="21.17" y2="8"></line>
+        <line x1="7.38" y1="12" x2="13.12" y2="2.06"></line>
+        <line x1="9.69" y1="16" x2="3.95" y2="6.06"></line>
+        <line x1="14.31" y1="16" x2="2.83" y2="16"></line>
+        <line x1="16.62" y1="12" x2="10.88" y2="21.94"></line>
+    </svg>
+);
+
+class HelloWorldPlugin {
+    initialize(registry, store) {
+        registry.registerChannelHeaderButtonAction(
+            // icon - JSX element to use as the button's icon
+            <ApertureIcon />,
+            // action - a function called when the button is clicked, passed the channel and channel member as arguments
+            // null,
+            () => { 
+                alert("Hello World!"); 
+            },
+            // dropdown_text - string or JSX element shown for the dropdown button description
```

Comment:
This parameter isn't actually working for me -- the icon still shows up in mobile view, so I need to investigate further here.

Reply:
The dropdown text is only used if enough plugins (2 right now) register channel header buttons. Otherwise it's unused

Reply:
Ah, interesting. That seems to present a bit of a challenge in styling, given the mobile blue background of the channel header. You can actually see this on pre-release too with the Zoom plugin shifting due to some CSS issue:

![image](https://user-images.githubusercontent.com/1023171/42885013-e08e450a-8a6d-11e8-9014-50311933ee82.png)

![image](https://user-images.githubusercontent.com/1023171/42885030-e9e1a4d0-8a6d-11e8-851c-25436044b1d2.png)

I've filed https://mattermost.atlassian.net/browse/MM-11386 to tackle this oddity. I'll update the text here to reflect reality.